### ISSUE TITLE
NaN in simValues_atExpTime

### ISSUE THREAD

**Issue by Christoph27** [Open-Systems-Pharmacology/Matlab-Toolbox#56]
In function "getPIWeightedResidualsForOSPSuiteExport.m":

Line:
simValues_atExpTime=interp1(sim_time,simValues,PI.output(iO).data_time);

If there are experimental values later than the last simulation time point, there is a NaN in the last element of "simValues_atExpTime". This gives a NaN in the residuals and in the total error at the end. 
I think the last NaN in "simValues_atExpTime" should just be discarded in such a case (there are legitimate situations to simulate shorter than all experimental data in a PI ;-).

**Comment by msevestre** [comment]
>there are legitimate situations to simulate shorter than all experimental data in a PI ;-).
lol

Interesting. I wonder if this (simulating shorter than all experimental data) works in PK-Sim /MoBi

**Comment by Christoph27** [comment]
Yes it works. In MoBi the PI went fine with reasonable errors. Only after exporting to Matlab I've got NaN as error.

**Comment by Christoph27** [comment]
... and I hope the experimental data after last simulation time point 
are not used somehow in PK-Sim/MoBi ...

**Comment by msevestre** [comment]
No. I am pretty sure this is all nicely done. I'll double check and paste the link to the code just for completion sake

**Comment by msevestre** [comment]
This is the line in code [here](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/890bee8ce5ee715ad8b8336f5d2aa6dc7558936b/src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ResidualCalculator.cs#L93)

and the test [When_calculating_the_residuals_with_observed_data_containing_containing_values_outside_of_the_simulation_range](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/890bee8ce5ee715ad8b8336f5d2aa6dc7558936b/tests/OSPSuite.Core.Tests/Core/ResidualCalculatorSpecs.cs#L576)

All good in PK-Sim/MoBi
